--- configure.ac	2018-09-13 10:05:59.923257708 -0400
+++ configure.ac.new	2018-09-13 10:09:47.731931566 -0400
@@ -13,7 +13,8 @@
 AC_CONFIG_FILES([src/include/switch_version.h.in:src/include/switch_version.h.template])
 
 AC_CONFIG_AUX_DIR(build/config)
-AM_INIT_AUTOMAKE
+#AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([subdir-objects])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 AC_CONFIG_SRCDIR([src/switch.c])
 AC_CONFIG_HEADER([src/include/switch_private.h])
@@ -1475,6 +1476,20 @@
       AC_MSG_ERROR([You need to either install libedit-dev (>= 2.11) or configure with --disable-core-libedit-support])
       ])])])
 
+dnl DH: Added for including google protobuf libs
+AC_ARG_WITH(grpc,
+  [AS_HELP_STRING([--with-grpc],
+    [enable support for google rpc (libgrpc++ and libgrpc)])],
+  [with_grpc="$withval"],
+  [with_grpc="no"])
+if test "$with_grpc" = "yes"; then
+  PKG_CHECK_MODULES([GRPC], [grpc++ grpc], [
+    AM_CONDITIONAL([HAVE_GRPC],[true])], [
+    AC_MSG_RESULT([no]); AM_CONDITIONAL([HAVE_GRPC],[false])])
+else
+  AM_CONDITIONAL([HAVE_GRPC],[false])
+fi
+
 dnl ---------------------------------------------------------------------------
 dnl - OpenLDAP SDK
 dnl ---------------------------------------------------------------------------

